cloudify/workflows/workflow_context.py

```diff
@@ -234,6 +234,21 @@ def __init__(self, ctx, node, node_instance, nodes_and_instances):
 
         self._logger = None
 
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            return False
+        return self.id == other.id
```

Comment:
Shouldn't this take into account other attributes in case of things like dep modifications?

Reply:
No, why? :)

Reply:
I've said in a docstring what I think it should do. But anyway, this way around it's more useful